teps/0073-simplify-metrics.md

```diff
@@ -0,0 +1,427 @@
+
+No need to outline all of the test cases, just the general strategy.  Anything
+that would count as tricky in the implementation and anything particularly
+challenging to test should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations).
+-->
+
+## Design Evaluation
+<!--
+How does this proposal affect the reusability, simplicity, flexibility
+and conformance of Tekton, as described in [design principles](https://github.com/tektoncd/community/blob/master/design-principles.md)
+-->
+
+## Drawbacks
+
+<!--
+Why should this TEP _not_ be implemented?
+-->
+
+## Alternatives
+
+
+### Alternatives 1: Customize metrics
+
+End users can also be allowed to customize what they want to collect and what level of granularity. We can add a flag to specify whether users want to customize the metrics.
+If it is set to `true`, Tekton pipelines would only report the metrics users configured.
+
+The configuration could be like this:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: config-observability
+  namespace: tekton-pipelines
+  labels:
+    app.kubernetes.io/instance: default
+    app.kubernetes.io/part-of: tekton-pipelines
+data:
+  metrics.customize-metrics: "true"
+  metrics.customize-metrics-spec: |
+    {
+      "metrics": [{
+          "name": "pipelinerun_duration_seconds",
+          "labels": ["status", "namespace"]
+        }, {
+          "name": "taskrun_duration_seconds",
+          "labels": ["status", "namespace"]
+        }, {
+          "name": "taskrun_count"
+        }
+      ]
+    }
+```
+
+If user doesn't specify the labels, use the default labels for the metrics. Besides, specifying every metrics users want might be annoying. We can only allow user to configure metrics whose type is histogram.
+
+**Drawbacks** includes:
+
+- Might be laborsome for users to configure what they want and influence the user experience
+- Take some effort to valid users input
+
+### Alternatives 2: Configure monitoring system filter
+
+Prometheus and Stackdriver both can filter the metrics by label. Users can customize condition configuration to choose what they need. We can provide a filter sample and let users configure monitoring system as they wish.
+
+**Drawbacks** includes:
+
+- Might slow down the monitoring system and cause new performance problem
+- Some monitoring system may not support label filter
+
+
+## Infrastructure Needed (optional)
+
+<!--
+Use this section if you need things from the project/SIG.  Examples include a
+new subproject, repos requested, github details.  Listing these here allows a
+SIG to get the process for these resources started right away.
+-->
+
+## Upgrade & Migration Strategy (optional)
+
+<!--
+Use this section to detail wether this feature needs an upgrade or
+migration strategy. This is especially useful when we modify a
+behavior or add a feature that may replace and deprecate a current one.
+-->
```

Comment:
I think this needs to be filled in.

Do we need to continue to support the current behaviour? 
Will it be opt-in or opt-out? Does that comply with our stability policy?

Reply:
From what I see in docs, metrics are still in experimental. We can support current metrics for one release by deprecating it and then removing it.